Cannot build TypeScript OOTB Echo BOT

### [Github issues](https://github.com/Microsoft/botbuilder-js/issues) should be used for bugs and feature requests. Use [Stack Overflow](https://stackoverflow.com/questions/tagged/botframework) for general "how-to" questions. 

## Versions
What package version of the SDK are you using. - 4.15.0
What nodejs version are you using - 14.15.4
What browser version are you using - none
What os are you using - Windows Server 2019 Datacenter 

## Describe the bug
When using Yeoman to create a TypeScript Echo BOT the following errors are displayed:

```sh
81 export declare type Resolved<T> = T extends {
                       ~~~~~~~~

node_modules/@azure/core-rest-pipeline/node_modules/@azure/core-tracing/types/core-tracing.d.ts:83:41 - error TS2315: Type 'Resolved' is not generic.

83 } ? F extends (value: infer V) => any ? Resolved<V> : never : T;
                                           ~~~~~~~~~~~

node_modules/@azure/core-rest-pipeline/node_modules/@azure/core-tracing/types/core-tracing.d.ts:131:212 - error TS2315: Type 'Resolved' is not generic.

131     }, Callback extends (updatedOptions: Options, span: Omit<TracingSpan, "end">) => ReturnType<Callback>>(name: string, operationOptions: Options, callback: Callback, spanOptions?: TracingSpanOptions): Promise<Resolved<ReturnType<Callback>>>;
     
```

## To Reproduce
Steps to reproduce the behavior:
1. yo botbuilder
2. What's the name of your bot? (my-chat-bot)
3. What will your bot do? (Demonstrate the core capabilities of the Microsoft Bot Framework)
4. What programming language do you want to use? **TypeScript**
5. Which template would you like to start with? (Use arrow keys) Echo Bot - https://aka.ms/bot-template-echo
6. Looking good.  Shall I go ahead and create your new bot? (Y/n)

## Expected behavior
Completes without errors.

## Screenshots

## Additional context

--- a/generators/generator-botbuilder/generators/app/templates/core/package-with-tests.json.ts
+++ b/generators/generator-botbuilder/generators/app/templates/core/package-with-tests.json.ts
@@ -54,6 +54,6 @@
       "nyc": "^15.0.1",
       "ts-node": "^8.10.1",
       "tslint": "^6.1.2",
-      "typescript": "^3.9.2"
+      "typescript": "^4.0.7"
   }
 }

--- a/generators/generator-botbuilder/generators/app/templates/core/package.json.ts
+++ b/generators/generator-botbuilder/generators/app/templates/core/package.json.ts
@@ -34,6 +34,6 @@
         "nyc": "^15.0.1",
         "ts-node": "^8.10.1",
         "tslint": "^6.1.2",
-        "typescript": "^3.9.2"
+        "typescript": "^4.0.7"
     }
 }

--- a/generators/generator-botbuilder/generators/app/templates/echo/package.json.ts
+++ b/generators/generator-botbuilder/generators/app/templates/echo/package.json.ts
@@ -27,6 +27,6 @@
         "@types/restify": "8.4.2",
         "nodemon": "^2.0.4",
         "tslint": "^6.1.2",
-        "typescript": "^3.9.2"
+        "typescript": "^4.0.7"
     }
 }

--- a/generators/generator-botbuilder/generators/app/templates/empty/package.json.ts
+++ b/generators/generator-botbuilder/generators/app/templates/empty/package.json.ts
@@ -26,6 +26,6 @@
         "@types/restify": "8.4.2",
         "nodemon": "^2.0.4",
         "tslint": "^6.1.2",
-        "typescript": "^3.9.2"
+        "typescript": "^4.0.7"
     }
 }

